gensim/models/ldamodel.py

```diff
@@ -631,8 +631,13 @@ def rho():
             return pow(offset + pass_ + (self.num_updates / chunksize), -decay)
 
         if self.callbacks:
+            # pass the list of input callbacks to Callback class
             callback = Callback(self.callbacks)
             callback.set_model(self)
+            # initialize metrics dict to store metric values after every epoch
+            self.metrics = {}
+            for metric in self.callbacks:
```

Comment:
Dict comprehension more readable?

Also, `defaultdict` might make the logic a little simpler.